src/braket/circuits/circuit.py

```diff
@@ -643,6 +651,61 @@ def add_verbatim_box(
             self._has_compiler_directives = True
         return self
 
+    def measure(self, target_qubits: np.ndarray | int = []) -> Circuit:
```

Comment:
Should `circuit.measure()` do measure all qubits?

Also Python has some [weird behavior](https://www.codiga.io/blog/python-avoid-empty-array-parameter/) when using mutable objects as default args, so we should update this signature to use None

Reply:
If there is no target specified, it would make sense to measure all the qubits on the circuit before the measure instruction. I was raising an error if there was no target specified but I think measuring all the qubits makes more sense.

Reply:
Updated the signature to use `None` instead of `[]` and updated to measure all qubits if no targets are provided.